catalog/controller/payment/razorpay.php

```diff
@@ -105,6 +105,17 @@ public function index(): string {
                 $data['is_recurring'] = "false";
                 // Orders API with payment autocapture
                 $order_data = $this->get_order_creation_data($this->session->data['order_id']);
+
+                if ($order_info['order_status_id'])
+                {
+                    $rzpOrder = $this->api->order->fetch($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]);

```

Comment:
@abdulwahidsharief , please apply the check for OrderID validation before using it for API.